### ISSUE TITLE
Searching with no results causes a warning

### ISSUE THREAD

**Issue by mariari** [atlas-engineer/nyxt#1489]
**Describe the bug**

When Searching a buffer with no results the following warning is given

```lisp
<WARN> [00:50:27] Warning: Error on separate thead: The value
                           -1
                         is not of type
                           UNSIGNED-BYTE
```

Seems rather benign but would be good to fix up.

turning

`(prompter:follow-p t)` to `(prompter:follow-p nil)` fixes it, so the bug lies somewhere in there.

**Precise recipe to reproduce the issue**

run `search-buffer` then press the entry down key.

**Information**
- OS name+version:
- Graphics card and driver:
- Desktop environment / Window manager name+version:
- How you installed Nyxt (Guix pack, package manager, build from source):
- Information from command copy-system-information:

  If you can't run copy-system-information, try `nyxt --system-information` from
  a shell.  It this still does not work, please provide the following:
  - Nyxt version (from =M-x nyxt-version= or =nyxt --version=):
  - Lisp implementation/version (if built from source):
  - Kernel name+version:
  - WebKitGTK+ | QtWebEngine version:

**Output when started from a shell**

**Comment by aartaka** [comment]
Indeed. I can reproduce. 75317372c30bed4e075c12ace676b79c5b693566 should fix that. However, I don't particularly like the solution I've gone with. @Ambrevar, can we bound-check `selection` better here? Maybe some math error that resulted in `selection` being `(something -1)` for a movement on step down in an empty source (sounds like an edge-case)?

**Comment by Ambrevar** [comment]
Your fix is fine I think.

But right, the index should never be negative.  I've fixed it in ada15ddd1507186440b7894678818f634c3b872e.  I've also added some type-checking to be more future-proof, considering how critical this function is.
